### PR TITLE
feat(compute): samples/compute_disk_list sample

### PR DESCRIPTION
**Pull request by irudykss**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/disks/list.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+// listDisks lists disks in a project zone.
+func listDisks(w io.Writer, projectID, zone, filter string) error {
+	// projectID := "your_project_id"
+	// zone := "us-central1-a"
+	// filter := ""
+
+	// Formatting for filters:
+	//https://cloud.google.com/python/docs/reference/compute/latest/google.cloud.compute_v1.types.ListDisksRequest
```
Comment (muncus) [inline compute/disks/list.go:34]: please use a golang link here, instead of python.

File: compute/disks/disks_test.go [hunk 2]
```diff
@@ -465,4 +465,40 @@ func TestComputeDisksSnippets(t *testing.T) {
 		// Cannot clean up the disk just yet because it must be done after the VM is terminated.
 		// It will be done by deleteInstance function.
 	})
+	t.Run("Create a test disk and list it", func(t *testing.T) {
```
Comment (muncus) [inline compute/disks/disks_test.go:468]: nit: this test does not create a disk. a more accurate comment would be "list disks"

File: compute/disks/disks_test.go [hunk 3]
```diff
@@ -465,4 +465,40 @@ func TestComputeDisksSnippets(t *testing.T) {
...
+		var buf bytes.Buffer
+
+		if err := listDisks(&buf, tc.ProjectID, zone, ""); err != nil {
+			t.Fatalf("listDisks got err: %v", err)
```
Comment (muncus) [inline compute/disks/disks_test.go:472]:
```suggestion
			t.Errorf("listDisks got err: %v", err)
```
while it is true this test fails on an error here, we can still continue. so this should be `Errorf` instead. The same is true for the two Fatalfs below.

**Review by muncus — COMMENTED** [review]
minor changes needed, but overall this looks good, thanks!

File: compute/disks/disks_test.go (hunk 2 above)
Comment (irudykss) [reply compute/disks/disks_test.go]: fixed

File: compute/disks/disks_test.go (hunk 3 above)
Comment (irudykss) [reply compute/disks/disks_test.go]: fixed

File: compute/disks/list.go (hunk 1 above)
Comment (irudykss) [reply compute/disks/list.go]: fixed

File: compute/disks/list.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+// listDisks lists disks in a project zone.
+func listDisks(w io.Writer, projectID, zone, filter string) error {
+	// projectID := "your_project_id"
+	// zone := "us-central1-a"
+	// filter := ""
+
+	// Formatting for filters:
+	// https://cloud.google.com/php/docs/reference/cloud-compute/latest/V1.ListDisksRequest
```
Comment (muncus) [inline compute/disks/list.go:34]: this is still the wrong link.

**Comment by muncus** [comment]
Please avoid force-pushing to PR branches, as it makes it much harder to identify which parts of the PR have been changed since the last review.

doing a `git pull` prior to your push should avoid the errors that prevented a non-force push.

File: compute/disks/list.go (hunk 4 above)
Comment (irudykss) [reply compute/disks/list.go]: lol, missed that, thanks 👍 
added inline explanation 👍

File: compute/disks/list.go [hunk 5]
```diff
@@ -31,7 +31,7 @@ func listDisks(w io.Writer, projectID, zone, filter string) error {
...
...u can use filtering on nested fields to filter based on resource labels. To filter on multiple expressions, provide each separate expression within parentheses. For example: ``` (scheduling.automaticRestart = true) (cpuPlatform = "Intel Skylake") ``` By default, each expression is an `AND` expression. However, you can include `AND` and `OR` expressions explicitly. For example: ``` (cpuPlatform = "Intel Skylake") OR (cpuPlatform = "Intel Broadwell") AND (scheduling.automaticRestart = true) ``` If you want to use a regular expression, use the `eq` (equal) or `ne` (not equal) operator against a single un-parenthesized expression with or without quotes or against multiple parenthesized expressions. Examples: `fieldname eq unquoted literal` `fieldname eq 'single quoted literal'` `fieldname eq "double quoted literal"` `(fieldname1 eq literal) (fieldname2 ne "literal")` The literal value is interpreted as a regular expression using Google RE2 library syntax. The literal value must match the entire field. For example, to filter for instances that do not end with name "instance", you would use `name ne .*instance`. You cannot combine constraints on multiple fields using regular expressions.
```
[the remark this answers is not in the document]
Comment (irudykss) [reply compute/disks/list.go:34]: NP 👍  Changed to python link

**Review by muncus — APPROVED** [review]

**Pull request merged**